src/libertem/io/dataset/k2is.py

```diff
@@ -567,10 +583,14 @@ def _do_initialize(self):
     def initialize(self, executor):
         return executor.run_function(self._do_initialize)
 
-    def _get_scansize(self):
-        with dm.fileDM(_get_gtg_path(self._path), on_memory=True) as dm_file:
-            return (int(dm_file.allTags['.SI Dimensions.Size Y']),
-                    int(dm_file.allTags['.SI Dimensions.Size X']))
+    def _get_frame_count(self):
+        with self._get_syncer().sectors[0] as sector:
+            first_block = sector.first_block()
+            last_block = sector.last_block()
+        first_frame_id = int(first_block.header['frame_id'])
+        last_frame_id = int(last_block.header['frame_id'])
+        frame_count = last_frame_id - first_frame_id
+        return (frame_count,)
```

Comment:
To make this re-usable (and to conform to its name), this could return an `int` instead of a `tuple`

Reply:
Yup, thanks.